client/my-sites/checkout/composite-checkout/payment-methods/credit-card/credit-card-pay-button.tsx

```diff
@@ -143,21 +142,26 @@ function ButtonContents( {
 	activeButtonText?: ReactNode;
 } ) {
 	const { __ } = useI18n();
+	const isPurchaseFree = total.amount.value === 0;
 	if ( formStatus === FormStatus.SUBMITTING ) {
 		return <>{ __( 'Processing…' ) }</>;
 	}
+	if ( formStatus === FormStatus.READY && isPurchaseFree ) {
+		const defaultText = (
+			<CreditCardPayButtonWrapper>{ __( 'Complete Checkout' ) }</CreditCardPayButtonWrapper>
+		);
+		/* translators: %s is the total to be paid in localized currency */
+		return <>{ activeButtonText || defaultText }</>;
```

Comment:
Interesting pattern, I was expecting `return <>{activeButtonText ?? defaultText}</>;` 👌🏾

Reply:
They're similar. In this case, using `||` will also use `defaultText` if `activeButtonText` is anything falsy, like an empty string. 

```
const activeButtonText = ''
const defaultText = 'default';
console.log( activeButtonText || defaultText ); // prints 'default'
console.log( activeButtonText ?? defaultText ); // prints ''
```